### PR TITLE
[feature] [comment] [frontend] adding flow

### PR DESCRIPTION
**Pull request by badukwei**
## Summary

Create adding comment flow for users in frontend.

## Linked Issue

https://github.com/social-tw/social-tw-website/issues/123

## Details

Any implementation detail worth pointing out, if any.

## Impacted Areas

List the files or modules that this PR will change.

## Tests

Mention all types of tests (unit, integration, performance, etc.) that you've carried out and if any specific environment or configuration is needed to run them.

## Possible Impacts

Describe any existing features that might be affected by this change, or any potential backward compatibility issues, if any.

## Visual Materials

If your changes have UI effects, consider attaching screenshots or videos to demonstrate them.

## Verification Steps

Provide a step-by-step guide for others to reproduce your results and verify your changes.

## Todo

Break down the progress of the PR for everyone to see what else you intend to include in this PR.

-   [ ] Draft implementation for `function`
-   [ ] Unit tests for `function`

## Checklist

-   [ ] All new and existing tests pass
-   [ ] I have updated the documentation (if applicable)
-   [ ] My changes do not introduce new security vulnerabilities
-   [ ] Run `yarn lint:fix`

### CODE REVIEW

File: packages/frontend/src/pages/PostDetail.tsx [hunk 1]
```diff
@@ -18,9 +20,25 @@ const demoPost = {
...
     const [post, setPost] = useState<PostInfo>()
 
+    const onSubmit = async (values: PostValues) => {
+        try {
+            console.log(values.content)
+            setIsModalOpen(true)
+            // TODO: await transactions
+            setTimeout(() => {
+                setIsModalOpen(false)
+                setIsModalOpen(false)
```
[the remark this answers is not in the document]
Comment (badukwei) [reply packages/frontend/src/pages/PostDetail.tsx:35]: I will remove it.

File: packages/frontend/src/components/post/PostForm.tsx [hunk 2]
```diff
@@ -15,13 +16,15 @@ export default function PostForm({
...
     disabled = false,
+    type = 'post'
 }: {
     onCancel?: () => void
     onSubmit?: (values: PostValues) => void
     onSubmitCancel?: () => void
     isSubmitCancellable?: boolean
     isSubmitCancelled?: boolean
     disabled?: boolean
+    type?: 'post' | 'comment'
```
[the remark this answers is not in the document]
Comment (badukwei) [reply packages/frontend/src/components/post/PostForm.tsx:27]: 我本來有建一個新的CommentForm，後來想說基本一樣就不要重複做了，如果分開比較好那我新建一個，我在想關掉 CommentForm 應該是點擊取消，還是點到 Form 的外面就會關掉。

**Comment by badukwei** [comment]
我不太確定現在開 pr 是要基本大致完成，還是就先建起來建起來同步進度，你覺得怎樣比較好呢 ? @Xiawpohr

**Comment by Xiawpohr** [comment]
我覺得可以先建起來同步進度，我會安排時間上來看 PR 上修正的程式碼 :)

File: packages/frontend/src/components/post/PostForm.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Xiawpohr) [reply packages/frontend/src/components/post/PostForm.tsx:27]: 我個人覺得點擊取消再關掉 CommnetForm，體驗上可能會比較好一點

**Review by Xiawpohr — APPROVED** [review]
LGTM

**Pull request merged**